lib/broadway_rabbitmq/channel_pool.ex

```diff
@@ -0,0 +1,20 @@
+defmodule BroadwayRabbitMQ.ChannelPool do
```

Comment:
I included only `checkout_channel/1` and `checkin_channel/1` on this behaviour because they're the only function that is called inside `AMQPClient`, I think the `init/1` callback is an implementation detail that doesn't need to be defined by this behaviour.